MERLin/MERLin/Module.swift

```diff
@@ -81,19 +99,29 @@ public extension AnyModule {
         guard rootViewController == nil else { return rootViewController! }
         let controller = unmanagedRootViewController()
         rootViewController = controller
-        
         _viewControllerEvent.onNext(.initialized)
-        let didAppearProducer = controller.rx.sentMessage(#selector(UIViewController.viewDidAppear(_:)))
+        bindViewController(controller, to: _viewControllerEvent)
+        return controller
+    }
+    
+    private func bindViewController(_ viewController: UIViewController, to events: BehaviorSubject<ViewControllerEvent>) {
+        let didAppearProducer = (viewController as UIViewController).rx.sentMessage(#selector(UIViewController.viewDidAppear(_:)))
             .map { _ in ViewControllerEvent.appeared }
         
-        let didDisappearProducer = controller.rx.sentMessage(#selector(UIViewController.viewDidDisappear(_:)))
+        let didDisappearProducer = (viewController as UIViewController).rx.sentMessage(#selector(UIViewController.viewDidDisappear(_:)))
             .map { _ in ViewControllerEvent.disappeared }
         
         Observable.of(didAppearProducer, didDisappearProducer)
             .merge()
-            .bind(to: _viewControllerEvent)
+            .bind(to: events)
             .disposed(by: disposeBag)
```

Comment:
We should test if internal VCs are being released in this way I'm not sure if `rx.sentMessage` is retaining. The disposeBag used here is the one coming from the module so VCs might be retained as long as the Module is alive. It's ok for the main VC but I don't think this is the behaviour we want for other VCs.